src/algorithms/controllers/PredictiveStanleyController.cpp

```diff
@@ -142,48 +138,71 @@ namespace controllers
-            */
-
-            // Get the reference path vector: from the closest waypoint to the next waypoint.
-            double dForwardVectorX = m_vReferencePath[m_nCurrentReferencePathTargetIndex + 1].GetUTMCoordinate().dEasting - stClosestWaypoint.GetUTMCoordinate().dEasting;
-            double dForwardVectorY =
-                m_vReferencePath[m_nCurrentReferencePathTargetIndex + 1].GetUTMCoordinate().dNorthing - stClosestWaypoint.GetUTMCoordinate().dNorthing;
-            // Compute the norm and unit vector for the path segment.
-            double dForwardNorm = sqrt(dForwardVectorX * dForwardVectorX + dForwardVectorY * dForwardVectorY);
-            double dFwdUnitX    = dForwardVectorX / dForwardNorm;
-            double dFwdUnitY    = dForwardVectorY / dForwardNorm;
-            // Get the vehicle's position vector relative to the closest waypoint.
+            geoops::Waypoint stClosestWaypoint = FindClosestWaypointInPath(stPredictedPosition);
+
+            // Compute the path forward vector from the segment start -> next waypoint (use index set by FindClosestWaypointInPath).
+            int nIdx                                    = m_nCurrentReferencePathTargetIndex;
+            int nNextIdx                                = std::min(nIdx + 1, static_cast<int>(m_vReferencePath.size() - 1));
+            const geoops::UTMCoordinate& stSegmentStart = m_vReferencePath[nIdx].GetUTMCoordinate();
+            const geoops::UTMCoordinate& stSegmentEnd   = m_vReferencePath[nNextIdx].GetUTMCoordinate();
+
+            // Forward vector of the path segment.
+            double dForwardVectorX = stSegmentEnd.dEasting - stSegmentStart.dEasting;
+            double dForwardVectorY = stSegmentEnd.dNorthing - stSegmentStart.dNorthing;
+            double dForwardNorm    = std::hypot(dForwardVectorX, dForwardVectorY);
+            // Degenerate segment. Skip this prediction step.
+            if (dForwardNorm < 1e-9)
+            {
+                continue;
+            }
+
+            // Unit forward vector.
+            double dFwdUnitX = dForwardVectorX / dForwardNorm;
+            double dFwdUnitY = dForwardVectorY / dForwardNorm;
+            // Math angle of segment (atan2 gives angle from +X (East), CCW positive) in degrees.
+            double dSegmentMathDeg = std::atan2(dForwardVectorY, dForwardVectorX) * (180.0 / M_PI);
+            // Convert math angle to COMPASS heading (0 = North, clockwise positive):
+            // compass = (90 - math) mod 360.
+            double dPathHeadingDeg = std::fmod(90.0 - dSegmentMathDeg + 360.0, 360.0);
+            // Predicted theta is produced by UnicycleModel in DEGREES (compass). Treat as degrees.
+            double dPredThetaDeg = dPredictedTheta;
+            // Small helper: signed smallest-angle difference in degrees in (-180, 180].
+            auto AngleDiffDeg = [](double targetDeg, double sourceDeg) -> double
```

Comment:
Remove `auto` type since it isn't an approved type in the team development standards

```suggestion
            std::function<double(double, double)> AngleDiffDeg = [](double targetDeg, double sourceDeg) -> double
```